src/lib/loaders/loaders_with_authentication/exchange.ts

```diff
@@ -13,13 +21,49 @@ export default (accessToken, opts) => {
   )
 
   // This generates a token with a lifetime of 1 minute, which should be plenty of time to fulfill a full query.
-  const exchangeTokenLoader = gravityLoader(
+  const exchangeTokenLoader: () => Promise<{ token: string }> = gravityLoader(
     "me/token",
     { client_application_id: EXCHANGE_APP_ID },
     { method: "POST" }
   )
 
+  const exchangeGraphQLLoader = async <R = unknown>({
+    query,
+    variables,
+  }: GraphQLArgs): Promise<Record<string, R>> => {
+    const { token } = await exchangeTokenLoader()
+
+    const body = JSON.stringify({
+      query,
+      variables,
+    })
+
+    const response = await fetch(urljoin(EXCHANGE_API_BASE, "graphql"), {
+      method: "POST",
+      body,
+      headers: {
+        Authorization: `Bearer ${token}`,
+        "Content-Type": "application/json",
+      },
+    })
+
+    const json = await response.json()
+    const { data: exchangeData, errors: exchangeErrors } = json
+    verbose(json)
+
+    // If the causality request failed for some reason, throw its errors.
```

Comment:
#superminor this comment should probably say `exchange` instead of `causality` 😄